core/src/main/java/brooklyn/management/internal/AbstractManagementContext.java

```diff
@@ -443,6 +441,22 @@ public void setManagementNodeUri(URI uri) {
         return uri;
     }
     
+    @Override
+    public CatalogInitialization getCatalogInitialization() {
+        if (catalogInitialization!=null) return catalogInitialization;
+        synchronized (this) {
+            if (catalogInitialization!=null) return catalogInitialization;
+            CatalogInitialization ci = new CatalogInitialization();
+            setCatalogInitialization(ci);
+            return ci;
+        }
+    }
+    
+    @Override
+    public synchronized void setCatalogInitialization(CatalogInitialization catalogInitialization) {
+        if (catalogInitialization!=null) catalogInitialization.injectManagementContext(this);
```

Comment:
I'd much prefer us to fail if the user passes in null, rather than failing on an NPE in some unrelated code/thread later.

I'd do something like:

    this.catalogInitialization  = checkNotNull(catalogInitialization, "catalogInitialization");
    catalogInitialization.injectManagementContext(this);